mpcforces_extractor/datastructure/rigids.py

```diff
@@ -93,3 +99,143 @@ def get_subcase_id2part_id2force(self) -> Dict:
+        differences = [abs(ev - mean_of_two_smallest) for ev in eigenvalues]
+        outlier_index = np.argmax(differences)
+
+        # Select the eigenvector corresponding to the outlier eigenvalue
+        axis = eigenvectors[:, outlier_index]
+
+        # Handle degenerate cases (e.g., 2D plane of points)
+        if np.isclose(
+            sorted(eigenvalues)[0], 0, atol=1e-6
+        ):  # Detect near-zero variance
+            Logger().log_warn(
+                f"Element {self.element_id} has a 2D plane of nodes; "
+                f"axis will be the cross product of the two dominant eigenvectors"
+            )
+            zero_index = np.argmin(eigenvalues)
+            # cross the other indices
+            axis = np.cross(
+                eigenvectors[:, (zero_index + 1) % 3],
+                eigenvectors[:, (zero_index + 2) % 3],
+            )
+
+        axis /= np.linalg.norm(axis)  # Normalize the axis
+
+        # Save and return the axis
+        self.axis = axis.tolist()
+        self.__compute_diameter_length(shifted_points, axis)
+
+        return self.axis
+
+    def __compute_diameter_length(
+        self, shifted_points: np.array, axis: np.array
+    ) -> None:
+
+        # Project points onto the axis
+        projections = np.dot(shifted_points, axis)
+
+        # Compute length (range of projections along the axis)
+        min_proj = np.min(projections)
+        max_proj = np.max(projections)
+
+        # Compute distances from points to the axis
+        distances = np.linalg.norm(shifted_points - np.outer(projections, axis), axis=1)
+
+        self.length = max_proj - min_proj
+        self.diameter = 2 * np.max(distances)
+
+    def get_part_id2axial_radial_forces(self, subcase: Subcase) -> Dict:
+        """
+        This method is used to get the axial and radial forces for the cylindrical MPCs
+        """
+
+        part_id2forces = self.get_part_id2force(subcase)
+
+        if self.axis is None:
+            self.axis = self.__fit_axis_for_cylindrical()
+
+        # Convert axis to a NumPy array
+        axis = np.array(self.axis)
+
+        part_id2axial_radial_forces = {}
+        for part_id, forces in part_id2forces.items():
+            # Forces are in global coordinates; extract only the first 3 components, no moments
+            forces = np.array(forces[0:3])
+
+            # Compute axial force (projection onto the axis)
+            axial_force_magnitude = np.dot(forces, axis)
+            axial_force = axial_force_magnitude * axis
+
+            # Compute radial force (remaining force perpendicular to the axis)
+            radial_force = forces - axial_force
+
+            # Store results
+            part_id2axial_radial_forces[part_id] = {
+                "axial_force": axial_force.tolist(),
+                "radial_force": radial_force.tolist(),
+            }
+
+        return part_id2axial_radial_forces
+
+    def get_max_radial_force(self, subcase: Subcase) -> Tuple[int, float]:
```

Comment:
Also doubting the concept here..... why is it like this, that you have to give in the subcase for getting the max force. 
Later pressing calculations you might want to get the max per part.

So first of all - it should be subcase2part2maxradial force ?! And again that stupid chaining.  I have to get around this.